File: cs/Basic/Server.EventTypes/MachineJobEventNode.cs
// Copyright (c) Traeger Industry Components GmbH. All Rights Reserved.

namespace EventTypes
{
    using Opc.UaFx;

    internal class MachineJobEventNode : OpcEventNode
    {
        #region ---------- Private fields ----------

        private OpcPropertyNode<MachineJob> jobNode;

        #endregion

        #region ---------- Public constructors ----------

        public MachineJobEventNode(OpcName name)
            : base(name)
        {
            this.InitializeChildren();
        }

        public MachineJobEventNode(IOpcNode parent, OpcName name)
            : base(parent, name)
        {
            this.InitializeChildren();
        }

        #endregion

        #region ---------- Public properties ----------

        public OpcPropertyNode<MachineJob> Job
        {
            get => this.jobNode;
        }

        #endregion

        #region ---------- Protected properties ----------

        protected override OpcNodeId DefaultTypeDefinitionId
        {
            get => "ns=1;s=MachineJobEventType";
        }

        #endregion

        #region ---------- Private methods ----------

        private void InitializeChildren()
        {
            this.jobNode = new OpcPropertyNode<MachineJob>(
                    parent: this,
                    name: "1:Job",
                    value: null);

            this.AddChild(OpcContext.Empty, this.jobNode);
        }

        #endregion
    }
}

File: vb/Basic/Server.EventTypes/MachineJobEventNode.vb
'Copyright (c) Traeger Industry Components GmbH. All Rights Reserved.

Imports Opc.UaFx

Namespace EventTypes
    Friend Class MachineJobEventNode
        Inherits OpcEventNode

        '---------- Private fields ----------

        Private jobNode As OpcPropertyNode(Of MachineJob)

        '---------- Public constructors ----------

        Public Sub New(ByVal name As OpcName)
            MyBase.New(name)
            Me.InitializeChildren()
        End Sub

        Public Sub New(ByVal parent As IOpcNode, ByVal name As OpcName)
            MyBase.New(parent, name)
            Me.InitializeChildren()
        End Sub

        '---------- Public properties ----------

        Public ReadOnly Property Job As OpcPropertyNode(Of MachineJob)
            Get
                Return Me.jobNode
            End Get
        End Property

        '---------- Protected properties ----------

        Protected Overrides ReadOnly Property DefaultTypeDefinitionId As OpcNodeId
            Get
                Return "ns=1;s=MachineJobEventType"
            End Get
        End Property

        '---------- Private methods ----------

        Private Sub InitializeChildren()
            Me.jobNode = New OpcPropertyNode(Of MachineJob)(parent:=Me, name:="1:Job", value:=Nothing)
            Me.AddChild(OpcContext.Empty, Me.jobNode)
        End Sub
    End Class
End Namespace
